rotkehlchen/chain/evm/decoding/gearbox/gearbox_cache.py

```diff
@@ -0,0 +1,305 @@
+                    values=[lp_token],
+                )
+
+
+def read_gearbox_data_from_cache(str_chain_id: str) -> tuple[dict[ChecksumEvmAddress, Any]]:
+    """Reads gearbox pools and names from global db cache tables."""
+    pools: dict[ChecksumEvmAddress, Any] = {}
+    with GlobalDBHandler().conn.read_ctx() as cursor:
+        for pool_address in get_existing_pools(
+            cursor=cursor,
+            cache_type=CacheType.GEARBOX_POOL_ADDRESS,
+            chain_id=str_chain_id,
+        ):
+            if (pool_name := globaldb_get_unique_cache_value(
+                cursor=cursor,
+                key_parts=(CacheType.GEARBOX_POOL_NAME, pool_address, str_chain_id),
+            )) is None:
+                continue
+            if (farming_pool_token := globaldb_get_unique_cache_value(
+                cursor=cursor,
+                key_parts=(CacheType.GEARBOX_POOL_FARMING_TOKEN, pool_address, str_chain_id),
+            )) is None:
+                continue
+            if len(token_lp_addresses := globaldb_get_general_cache_keys_and_values_like(
+                cursor=cursor,
+                key_parts=(CacheType.GEARBOX_POOL_LP_TOKENS, pool_address, str_chain_id),
+            )) == ZERO:
+                continue
+            pools[pool_address] = GearboxPoolData(
+                pool_address=pool_address,
+                pool_name=pool_name,
+                farming_pool_token=farming_pool_token,
+                lp_tokens={addr[1] for addr in token_lp_addresses},  # just the token address
+                str_chain_id=str_chain_id,
+            )
+
+    return (pools,)
+
+
+def ensure_gearbox_tokens_existence(
+        evm_inquirer: 'EvmNodeInquirer',
+        all_pools: list[GearboxPoolData],
+) -> list[GearboxPoolData]:
+    """This function receives data about gearbox pools and ensures that lp tokens and pool coins
+    exist in rotki's database."""
+    verified_pools = []
+    for pool in all_pools:
+        # Ensure pool coins exists and underlying tokens in the globaldb.
+        if pool.underlying_token is not None:
+            try:
+                get_or_create_evm_token(
+                    userdb=evm_inquirer.database,
+                    evm_address=pool.underlying_token,
+                    chain_id=evm_inquirer.chain_id,
+                    evm_inquirer=evm_inquirer,
+                    encounter=TokenEncounterInfo(
+                        description='Querying gearbox pools',
+                        should_notify=False,
+                    ),
+                )
+            except NotERC20Conformant as e:
+                log.error(f'Skipping pool {pool} because {pool.underlying_token} is not a valid ERC20 token. {e}')  # noqa: E501
+                continue
+
+        # ensure the lp and farming tokens exist in the db
+        verified_tokens: defaultdict[str, set[str]] = defaultdict(set)
```

Comment:
Hmm the logic here is a bit complicated and with the pop down there it's hard to understand. Why do you need a set? I am not sure I follow. Cause a token may exist many times and you only want unique items?

Also verified tokens means what? Existing in the db? The name is a bit confusing